plutus-contract/src/Language/Plutus/Contract/Test.hs

```diff
@@ -122,11 +122,12 @@ not :: TracePredicate s e a -> TracePredicate s e a
 not = PredF . fmap (fmap Prelude.not) . unPredF
 
 checkPredicate
-    :: forall s e a.
-       String
+    :: forall s e a
```

Comment:
You could make it `forall e s a`, so that you don't have to write `@GameSchema @AssertionError` below and `@AssertionError` suffices.

Reply:
Yeah, I thought about this, but I want to keep the same order of type variables throughout, and generally I think it makes more sense for the schema to come first.